### ISSUE TITLE
AttributeError: module 'nltk' has no attribute 'download

### ISSUE THREAD

**Issue by 2hands10fingers** [nltk/nltk#1961]
```
#code 

import nltk
nltk.download()
```

```
Traceback (most recent call last):
  File "nltk.py", line 2, in <module>
    import nltk
  File "/Users/antonionogueras/Desktop/NLTK/nltk.py", line 10, in <module>
    nltk.download()
AttributeError: module 'nltk' has no attribute 'download'
```

Specs:
Mac, High Sierra, Python 3.6

**Comment by somnathrakshit** [comment]
Try putting the name of the model as a parameter of the download() function. e.g.

```
import nltk
nltk.download('punkt')
```

**Comment by 2hands10fingers** [comment]
@somnathrakshit I have done so by adding 'popular' and 'all', but receive the same error.

**Comment by somnathrakshit** [comment]
'all' and 'popular' will work in an interactive console. Are you trying that or running from a .py script?

**Comment by 2hands10fingers** [comment]
@somnathrakshit Yes. I also tried `nltk.download('punkt')` as  you suggested and got the same error.

**Comment by somnathrakshit** [comment]
Can you try it out from an interactive console instead of a .py script? It worked for me.

**Comment by alvations** [comment]
Please rename your file from `nltk.py` to something else, e.g. `my_nltk_script.py`. Then rerun the script. Please see 

- https://meta.stackoverflow.com/questions/360877/canonical-q-a-for-module-x-has-no-attribute-y-in-python 
- https://sopython.com/canon/96/attributeerror-module-object-has-no-attribute-foobar/

Also, normally you shouldn't be downloading the NLTK data downloads in a script, unless it's an interactive jupyter notebook or interactive interpreter.

**Comment by alvations** [comment]
@2hands10fingers did changing your script filename resolve the issue?

**Comment by 2hands10fingers** [comment]
@alvations While it did bring up the GUI as intended, I have run into an SSL error which has rendered my ability to interact with the GUI.  This is my error.

```
[SSL: CERTIFICATE_VERIFY_FAILED]
certificate verify failed(ssl.c:749)
```

**Comment by alvations** [comment]
Are you using Python3.6? If so, it looks like ~an upstream bug from CPython~ similar to this issue: https://bugs.python.org/issue28150 

See 

- https://stackoverflow.com/questions/41348621/ssl-error-downloading-nltk-data
- https://stackoverflow.com/a/41692664/610569


TL;DR
====

```
/Applications/Python 3.6/Install Certificates.command
```

**Comment by 2hands10fingers** [comment]
`sudo /Applications/Python\ 3.6/Install\ Certificates.command`

Did the trick! I was definitely using Python 3.6. Thank you!

**Comment by ghost** [comment]
This worked for me.
At first, I had saved a file named nltk.py and in that file, I had written those 2 lines.
Go to C:\Users\User\AppData\Local\Programs\Python\Python36\
In here, delete that nltk.py file.

Now, re-run those 2 lines from the cmd prompt and it'll work.

**Comment by ajaykumar-br** [comment]
It still isn't working for me I have tried all of the above ones and yet I still get 
"module 'nltk' has no attribute 'internals' "

**Comment by vaishukla25** [comment]
@2hands10fingers  the same problem was with me,i uninstall nltk the reinstall it again then again run those line of code.
1.pip uninstall nltk
2.pip install nltk
3.again run those lines import nltk nltk.download()
4.run it ,it will execute errorfree.

**Comment by futurespoir** [comment]
Your terminal Python installation and your ipython install are separate.
What does import sys; print(sys.path) produce in either?

**Comment by ghost** [comment]
I named my python script nltk.py and that was causing the error. Therefore, I changed it to a different filename and voila it started working!

**Comment by Senzokuhle** [comment]
> This worked for me.
> At first, I had saved a file named nltk.py and in that file, I had written those 2 lines.
> Go to C:\Users\User\AppData\Local\Programs\Python\Python36
> In here, delete that nltk.py file.
> 
> Now, re-run those 2 lines from the cmd prompt and it'll work.

This solved the issue for me, I am running Python 3.8 on windows 10

**Comment by sahilrevankar** [comment]
> Please rename your file from `nltk.py` to something else, e.g. `my_nltk_script.py`. Then rerun the script. Please see
> 
> * https://meta.stackoverflow.com/questions/360877/canonical-q-a-for-module-x-has-no-attribute-y-in-python
> * https://sopython.com/canon/96/attributeerror-module-object-has-no-attribute-foobar/
> 
> Also, normally you shouldn't be downloading the NLTK data downloads in a script, unless it's an interactive jupyter notebook or interactive interpreter.


This solved the issue for me

**Comment by morkev** [comment]
For newer NLTK versions, use:
```python
nltk.download('punkt_tab')
```

Instead of what you likely have, which is:
```python
nltk.download('punkt')
```
